### PR TITLE
Port du-doorbell endpoint implementation into Aroo

### PR DESCRIPTION
**Pull request by anaulin**
### What does this code do, and why?

This PR moves the implementation of the doorbell endpoints from a separate app into Aroo. This is one of the steps in the work for https://github.com/doubleunion/arooo/issues/360 .

The code is largely untouched (including temporarily commented-out code and such), except for changes to adapt Sinatra code to Rails, and changing from using a keycode dictionary to querying our database.

### How is this code tested?

Unittests. Light manual testing locally to verify that XML responses seem correct.

### Are any database migrations required by this change?

No.

### Screenshots (before/after)

n/a

### Are there any configuration or environment changes needed?

Yes. Arooo now depends on having a Redis instance available to make the doorbell code work. I have already provisioned a free Heroku Redis on both our staging and production environments.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,6 +71,7 @@ Do the below OR if you prefer docker, see the Docker Setup section
 1. `gem install bundler`
 1. Fork the repo (click the Fork button above), and clone your fork to your local machine. [Here's a GitHub tutorial](https://help.github.com/articles/fork-a-repo/)
 1. Make sure that postgres is installed [brew install postgres](https://wiki.postgresql.org/wiki/Homebrew) OR brew postgresql-upgrade-database (if you have an older version of postgres)
+1. If you want to run the doorbell code locally, you will need to have a local instance of [Redis](https://redis.io/). You can install it on mac with `brew install redis`.
```
Comment (compwron) [inline README.md:74]: What's the plan for configuring running redis in heroku to make this work in prod?

File: README.md [hunk 2]
```diff
@@ -189,6 +191,12 @@ Now you can update any user:
 
 If you need to make or unmake an admin, have a current admin click the un/make admin button on a member in the Member Admin View. Admins can accept/reject applications, update any member's status, see current member's dues, open and close applications, and manage new member setup.
 
+### Programmatic doorbell
+
+Arooo includes code to handle incoming voice calls and text messages from an intercom system. This allows members to enter a personalized door code to open the door to our space.
+
+This code is a [Twilio TwiML](https://www.twilio.com/docs/voice/twiml) app that lives in the [DoorbellController](app/controllers/doorbell_controller.rb).
```
Comment (compwron) [inline README.md:198]: Yay docs!! :)

File: spec/controllers/doorbell_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,143 @@
+require "spec_helper"
```
Comment (compwron) [inline spec/controllers/doorbell_controller_spec.rb:1]: Yay tests!! 💖

**Comment by compwron** [comment]
It looks very nice! I love the tests 💖
I'm not clear on how redis config will work in heroku, I'm sure its possible tho!
The next step might be showing a member their code & allowing changing it?
Loading all the existing codes into the db eventually will be a pain 😅
👏🏼✨💪✨

**Comment by anaulin** [comment]
@compwron I've already provisioned a Heroku Redis on both staging and production, and confirmed that this set the `REDIS_URL` env variable we need. I used the free Hobby Dev tier for Heroku Redis, which is the one we are currently using in the `du-doorbell` app. I think with that the connection to Redis should Just Work ™️ ©️ 🤞 .

**Comment by anaulin** [comment]
This is deployed to staging, and I was able to do a basic test that the endpoints are there and respond.

Try it yourself, with:
https://doubleunion-staging.herokuapp.com/doorbell/sms?Body=123456

(No actual codes are in the staging database, so it is not possible right now to test the "successful keycode" path.)

**Comment by anaulin** [comment]
This is not deployed to production. See, e.g: https://app.doubleunion.org/doorbell

**Comment by anaulin** [comment]
That previous comment meant to say that this is _now_ deployed to production. See, e.g: https://app.doubleunion.org/doorbell

**Pull request merged**